pennylane/operation.py

```diff
@@ -1291,6 +1292,8 @@ def __add__(self, other):
             return qml.op_sum(self, other)
         if other == 0:
             return self
+        if isinstance(other, qml.ops.ParametrizedHamiltonian):  # pylint: disable=no-member
+            return other.__add__(self)
```

Comment:
Can we go ahead and remove this now? @AlbertMitjans

Reply:
Would that mean not allowing two `ParametrizedHamiltonian` instances to be added together?

Reply:
No, this method is just about adding an `Operator` and a `ParametrizedHamiltonian`. 

The reason I think it can be removed (or can be soon) is because the `__add__` and `__radd__` methods are defined on `ParametrizedHamiltonian`. So far, it doesn't get to use the `ParametrizedHamiltonian.__radd__` method when adding `Operator + ParametrizedHamiltonian`, because it tried to do `Operator.__add__` first, which raises a ValueError, and it just stops there. My understanding is that Albert has opened a PR that changes this to a `NotImplementedError` instead, which will cause it to check for and `__radd__` method on `ParametrizedHamiltonian` when `Operator. __add__` fails. That way we don't have to make the change here, we only have to define how to add them together on `ParametrizedHamiltonian`. At least this is how I've understood it, but Albert should be able to say if this sounds right.

Reply:
Cool! thanks for the explanation, wasnt aware of that :)

Reply:
Yes! We should be able to remove this